core/src/androidMain/kotlin/com/revenuecat/purchases/kmp/Purchases.android.kt

```diff
@@ -1,309 +1,373 @@
-@file:Suppress("EXTENSION_SHADOWED_BY_MEMBER")
-
 package com.revenuecat.purchases.kmp
 
 import com.revenuecat.purchases.PurchaseParams
-import com.revenuecat.purchases.Purchases
+import com.revenuecat.purchases.common.PlatformInfo
 import com.revenuecat.purchases.getCustomerInfoWith
 import com.revenuecat.purchases.getOfferingsWith
 import com.revenuecat.purchases.getProductsWith
 import com.revenuecat.purchases.kmp.di.AndroidProvider
 import com.revenuecat.purchases.kmp.di.requireActivity
+import com.revenuecat.purchases.kmp.di.requireApplication
+import com.revenuecat.purchases.kmp.models.BillingFeature
 import com.revenuecat.purchases.kmp.models.GoogleReplacementMode
 import com.revenuecat.purchases.kmp.models.PromotionalOffer
 import com.revenuecat.purchases.kmp.models.StoreMessageType
 import com.revenuecat.purchases.kmp.models.StoreProduct
 import com.revenuecat.purchases.kmp.models.StoreProductDiscount
 import com.revenuecat.purchases.kmp.models.StoreTransaction
 import com.revenuecat.purchases.kmp.models.SubscriptionOption
+import com.revenuecat.purchases.kmp.strings.ConfigureStrings
 import com.revenuecat.purchases.logInWith
 import com.revenuecat.purchases.logOutWith
 import com.revenuecat.purchases.models.InAppMessageType
 import com.revenuecat.purchases.purchaseWith
 import com.revenuecat.purchases.restorePurchasesWith
 import com.revenuecat.purchases.syncAttributesAndOfferingsIfNeededWith
 import com.revenuecat.purchases.syncPurchasesWith
-import com.revenuecat.purchases.Purchases as RcPurchases
-
-public actual typealias Purchases = RcPurchases
-
-public actual var Purchases.finishTransactions: Boolean
-    get() = finishTransactions
-    set(value) {
-        finishTransactions = value
+import java.net.URL
+import com.revenuecat.purchases.DangerousSettings as AndroidDangerousSettings
+import com.revenuecat.purchases.Purchases as AndroidPurchases
+import com.revenuecat.purchases.hybridcommon.configure as commonConfigure
+
+public actual class Purchases private constructor(private val androidPurchases: AndroidPurchases) {
+    public actual companion object {
+
+        private var _sharedInstance: Purchases? = null
```

Comment:
Hmm I wonder if we should keep the instance here or just rely on the native layer... It would make the instance methods just call a static method, similar to how you are doing the `logHandler` I guess? Not sure, I think this is ok for now though!

Reply:
Hmm, but we would be creating new `Purchases` wrappers then every time the user calls `sharedInstance`? Or did you have something else in mind?  
  
This actually prompted me to change `logHandler` to also keep a reference. 😅 The reason being that the user will expect the `logHandler` field to hold exactly what they set it to. If we don't keep a reference, we are converting to and from the platform type on `set` and `get`. That means if they set it to an instance of `MyAwesomeLogHandler`, they expect to be able to get the same `MyAwesomeLogHandler` instance by reading `Purchases.logHandler`. If we are converting to and from the platform type, they will just receive an anonymous object implementing `LogHandler`.

Let me know if you disagree!

Reply:
Hmm right, if the instance is not the same it could lead to issues... Yeah I think it makes sense, let's keep it this way then!